cyctl/utility/describer.go

```diff
@@ -0,0 +1,142 @@
+package utility
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+type Describer struct {
+	Prefix string
+	out    *tabwriter.Writer
+	buf    *bytes.Buffer
+}
+
+func Describe(fn func(d *Describer)) string {
+	d := Describer{
+		out: new(tabwriter.Writer),
+		buf: new(bytes.Buffer),
+	}
+	d.out.Init(d.buf, 0, 8, 2, ' ', 0)
+
+	fn(&d)
+
+	d.out.Flush()
+	return d.buf.String()
+}
+
+func (d *Describer) Printf(msg string, args ...interface{}) {
+	fmt.Fprint(d.out, d.Prefix)
+	fmt.Fprintf(d.out, msg, args...)
+}
+
+func (d *Describer) Println(args ...interface{}) {
+	fmt.Fprint(d.out, d.Prefix)
+	fmt.Fprintln(d.out, args...)
+}
+
+// DescribeModuleMetaData describes standard Module metadata in a consistent manner.
+func (d *Describer) DescribeModuleMetaData(metadata v1alpha1.Module) {
+	d.Printf("Name:\t%s\n", metadata.Name)
+	d.Printf("Namespace:\t%s\n", metadata.Namespace)
+	d.DescribeMap("Labels", metadata.Labels)
+	d.DescribeMap("Annotations", metadata.Annotations)
+
+}
+
+// DescribeTemplateAuthMetaData describes standard TemplateAuthRule metadata in a consistent manner.
+func (d *Describer) DescribeTemplateAuthMetaData(metadata v1alpha1.TemplateAuthRule) {
+	d.Printf("Name:\t%s\n", metadata.Name)
+	d.Printf("Namespace:\t%s\n", metadata.Namespace)
+	d.DescribeMap("Labels", metadata.Labels)
+	d.DescribeMap("Annotations", metadata.Annotations)
+
+}
+
+// DescribeTemplateStoreMetaData describes standard TemplateStore metadata in a consistent manner.
+func (d *Describer) DescribeTemplateStoreMetaData(metadata v1alpha1.TemplateStore) {
+	d.Printf("Name:\t%s\n", metadata.Name)
+	d.Printf("Namespace:\t%s\n", metadata.Namespace)
+	d.DescribeMap("Labels", metadata.Labels)
+	d.DescribeMap("Annotations", metadata.Annotations)
+}
```

Comment:
Can you make a single method for metadata since all of them share the same metadata object
```suggestion
func (d *Describer) DescribeMetadata(metadata metav1.ObjectMeta) {
	d.Printf("Name:\t%s\n", metadata.Name)
	d.Printf("Namespace:\t%s\n", metadata.Namespace)
	d.DescribeMap("Labels", metadata.Labels)
	d.DescribeMap("Annotations", metadata.Annotations)
}
```

Reply:
yeah, sure